src/main/java/com/dynatrace/prototype/payloadCreator/GetSLIMapper.java

```diff
@@ -28,12 +31,25 @@
 
         if (eventDataObject instanceof KeptnCloudEventGetSLIData) {
             KeptnCloudEventGetSLIData eventData = (KeptnCloudEventGetSLIData) eventDataObject;
-            StringBuilder specificDataSB = new StringBuilder();
-
-            specificDataSB.append(ifNotNull("Sli Provider: ", eventData.getSliProvider(), "\n"));
+            StringBuilder message = new StringBuilder();
+            KeptnCloudEventDataResult result = eventData.getResult();
+            KeptnEvent eventType = KeptnEvent.valueOf(StringUtils.upperCase(event.getPlainEventType()));
+            String service = eventData.getService();
+            String stage = eventData.getStage();
+            String project = eventData.getProject();
+
+            if (stage == null) {
+                System.err.printf(ERROR_NULL_VALUE, "Stage", eventName);
+            } else if (service == null) {
+                System.err.printf(ERROR_NULL_VALUE, "Service", eventName);
+            } else if (project == null) {
+                System.err.printf(ERROR_NULL_VALUE, "Project", eventName);
```

Comment:
Inside `createMessage()`, there is a check for `null` values. 
I would change this `if-else` in a cascade of `if`s so that if all of them are `null`, we get three error messages. Furthermore, I would change `createMessage()` to return an empty string and not `null` by default -> this way it is not coerced into the string `"null"`